### PR TITLE
export feed/fetch variable method to Python

### LINKED ISSUE #4836: need to export feed/fetch variable method to Python

### PR DESCRIPTION
**Pull request by QiJune**
Fix #4836

### CODE REVIEW

File: paddle/framework/feed_fetch_method.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  // If var_name Variable is not found in GlobalScope, a new variable will
+  // be created.
+  Variable* g_feed_value = GetGlobalScope().Var(var_name);
+  auto& feed_inputs =
+      *(g_feed_value->GetMutable<std::vector<paddle::framework::LoDTensor>>());
+  if (index >= feed_inputs.size()) {
+    feed_inputs.resize(index + 1);
+  }
+  // shared data with input tensor
+  feed_inputs[index].ShareDataWith<T>(input);
```
Comment (Canpio) [inline paddle/framework/feed_fetch_method.h]: I tend to use `CopyFrom()` here, for I'm afraid users wish to feed different versions of the same tensor to different `index`.

Comment (tonyyang-svail) [reply paddle/framework/feed_fetch_method.h]: FeedOp will make a copy of the data [[code]](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/operators/feed_op.cc#L46), so I believe it is okay for it to be shared here.

File: paddle/framework/feed_fetch_method.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+namespace framework {
+
+template <typename T>
+void SetFeedVariable(const LoDTensor& input, const std::string& var_name,
+                     size_t index) {
+  // If var_name Variable is not found in GlobalScope, a new variable will
+  // be created.
+  Variable* g_feed_value = GetGlobalScope().Var(var_name);
+  auto& feed_inputs =
+      *(g_feed_value->GetMutable<std::vector<paddle::framework::LoDTensor>>());
```
Comment (reyoung) [inline paddle/framework/feed_fetch_method.h]: Please use global typedefs here.

```
#include "paddle/framework/feed_fetch_type.h"
```

File: paddle/framework/feed_fetch_method.h (hunk 1 above)
Comment (Canpio) [reply paddle/framework/feed_fetch_method.h]: I see. Thanks!

**Pull request merged**